packages/visualizations/src/components/Map/utils.ts

```diff
@@ -227,8 +229,78 @@ export const getFixedTooltips = (
     ) as ChoroplethFixedTooltipDescription[];
 };
 
-export const isVectorTile = (
-    shape: ChoroplethShapeValues
-): shape is ChoroplethShapeVectorTilesValue =>
-    (shape as ChoroplethShapeVectorTilesValue).layer !== undefined &&
-    (shape as ChoroplethShapeVectorTilesValue).key !== undefined;
+export const computeFilterExpression = (filterConfig: MapFilter) => {
+    /** Transform a filter object from the options into a Maplibre filter expression */
+    const { key, value } = filterConfig;
+    const filterMatchExpression: string[] = ['in', key];
+    (Array.isArray(value) ? value : [value]).forEach((filterValue) => {
+        filterMatchExpression.push(filterValue.toString());
+    });
+    return filterMatchExpression;
+};
+
+export const defaultFormat: ChoroplethTooltipFormatter = ({ value, label }) =>
+    value ? `${label} &mdash; ${value}` : label;
+
+export const computeTooltip: ComputeTooltipFunction = (
+    hoveredFeature,
+    dataValues,
+    options,
+    matchKey
+) => {
+    const values = dataValues || [];
+    const matchedFeature = values.find(
+        (item: ChoroplethDataValue) => String(item.x) === hoveredFeature.properties?.[matchKey]
+    );
+
+    let tooltipLabel = hoveredFeature.properties?.label || hoveredFeature.properties?.[matchKey];
+    const labelMatcher = options?.tooltip?.labelMatcher;
+
+    if (labelMatcher && matchedFeature) {
+        const { type } = labelMatcher;
+        if (type === 'keyProperty') {
+            const { key } = labelMatcher;
+            tooltipLabel = hoveredFeature.properties?.[key];
+        } else if (type === 'keyMap') {
+            const { mapping } = labelMatcher;
+            tooltipLabel = mapping[matchedFeature?.x];
+        }
+    }
+
+    const tooltipRawValues: {
+        value?: number;
+        label: string;
+        key: string;
+    } = {
+        value: matchedFeature?.y,
+        label: tooltipLabel,
+        key: hoveredFeature.properties?.[matchKey], // === matchedFeature.x
+    };
+    const format = options?.tooltip?.labelFormatter;
+
+    return format ? format(tooltipRawValues) : defaultFormat(tooltipRawValues);
+};
+
+export const computeBaseLayer = (
+    fillColor: string | (string | string[])[],
+    DefaultColor: Color
+): ChoroplethLayer => ({
+    type: 'fill',
+    layout: {},
+    paint: {
+        'fill-color': fillColor,
+        'fill-opacity': 0.8,
+        'fill-outline-color': DefaultColor,
+    },
+});
+
+export const computeMatchExpression = (
```

Comment:
I wonder if we could enforce the `FilterSpecification` from `maplibre` for the return value here. At first glance it seems to be intended for it. If it works, it would make further changes in this function more "secure".

Reply:
Yes Indeed, I've added it and it seems working